src/public/modules/forms/base/directives/field.client.directive.js

```diff
@@ -23,6 +25,28 @@ function fieldDirective(FormFields) {
       isValidateDate: '<',
     },
     link: function (scope) {
+      // Stealth prefill feature
+      // If a query parameter is provided to a form URL in the form ?<fieldId1>=<value1>&<fieldId2>=<value2>...
+      // And if the fieldIds are valid mongoose object IDs and refer to a short text field,
+      // Then prefill and disable editing the corresponding form field on the frontend
+
+      const decodedUrl = he.decode($location.url()) // tech debt; after redirect, & is encoded as &amp; in the query string
+      const query = decodedUrl.split('?')[1]
```

Comment:
can i confirm there's no null pointer exception if the URL doesn't contain `?`?

Reply:
or index out of bounds with the `[1]` index access

Reply:
yes, there is no null pointer exception because when we call [split](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/split), `If separator is omitted or does not occur in str, the returned array contains one element consisting of the entire string.`

accessing index out of bounds returns undefined, but since this isn't in the specifications i've added an array length check